libraries/networking/src/udt/SendQueue.cpp

```diff
@@ -0,0 +1,252 @@
+//
+//  SendQueue.cpp
+//  libraries/networking/src/udt
+//
+//  Created by Clement on 7/21/15.
+//  Copyright 2015 High Fidelity, Inc.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+#include "SendQueue.h"
+
+#include <algorithm>
+
+#include <QtCore/QCoreApplication>
+#include <QtCore/QThread>
+
+#include <SharedUtil.h>
+
+#include "ControlPacket.h"
+#include "Packet.h"
+#include "Socket.h"
+
+using namespace udt;
+using namespace std::chrono;
+
+std::unique_ptr<SendQueue> SendQueue::create(Socket* socket, HifiSockAddr dest) {
+    auto queue = std::unique_ptr<SendQueue>(new SendQueue(socket, dest));
+    
+    Q_ASSERT_X(socket, "SendQueue::create", "Must be called with a valid Socket*");
+    
+    // Setup queue private thread
+    QThread* thread = new QThread();
+    thread->setObjectName("Networking: SendQueue " + dest.objectName()); // Name thread for easier debug
+    
+    connect(thread, &QThread::started, queue.get(), &SendQueue::run);
+    
+    connect(queue.get(), &QObject::destroyed, thread, &QThread::quit); // Thread auto cleanup
+    connect(thread, &QThread::finished, thread, &QThread::deleteLater); // Thread auto cleanup
```

Comment:
if the Connection doesn't wait on this thread I'm not sure that cleanup will truly be seamless while app is going down - we may need to wait on this thread in the Connection dtor once SendQueue deleteLater is called